src/Cake.Common/Build/AppVeyor/AppVeyorProvider.cs

```diff
@@ -88,12 +98,35 @@ public void UploadArtifact(FilePath path)
             arguments.AppendQuoted(path.FullPath);
             arguments.Append("-FileName");
             arguments.AppendQuoted(path.GetFilename().FullPath);
+            arguments.Append("-ArtifactType");
+            arguments.AppendQuoted(settings.ArtifactType.ToString());
+            if (!string.IsNullOrEmpty(settings.DeploymentName))
+            {
+                if (settings.DeploymentName.Contains(" "))
+                {
+                    throw new CakeException("The deployment name can not contain spaces");
+                }
+                arguments.Append("-DeploymentName");
+                arguments.AppendQuoted(settings.DeploymentName);
+            }
 
             // Start the process.
             _processRunner.Start("appveyor", new ProcessSettings { Arguments = arguments });
```

Comment:
Random might be out of scope thought, should "appveyor.exe" be a tool? Currently we just fire and forget. Tool would validate exit code too.